vhdl_lang/src/ast/visitor.rs

```diff
@@ -327,6 +327,13 @@ pub trait Visitor {
     ) -> VisitorResult {
         Continue
     }
+    fn visit_subprogram_specification(
```

Comment:
I can remove it and skip the `SubprogramSpecification` if you want. The implementation for `ASTNode` is still needed, though, since `ProcedureSpecification` and `FunctionSpecification` did already have their own visit functions.

Reply:
You can remove those as well. I do not consider ASTNode final public API. I hope to rework it in the future to create a stable public API. Now we only added what was needed for internal use.

Reply:
I think you removed a bit much. We still need the ASTNode implementation of the specifications to visit the children.  I was just asking to remove the visit_subprogram_specification from the Visitor trait.